### PR TITLE
CC-29675 Shipment Type support on Checkout & OMS

### PR DESCRIPTION
**Pull request by AsonUnique**
#### Overview:

- Jira epic link: https://spryker.atlassian.net/browse/CC-29675
- Jira IG link: https://spryker.atlassian.net/browse/CC-30143

### CODE REVIEW

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-inventory-management-feature.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 
 
-This document describes how to ingrate the [Order Management](/docs/scos/user/features/{{page.version}}/order-management-feature-overview/order-management-feature-overview.html) + [Inventory Management](/docs/pbc/all/warehouse-management-system/{{page.version}}/base-shop/inventory-management-feature-overview.html) feature into a Spryker project.
+This document describes how to integrate the [Order Management](/docs/scos/user/features/{{page.version}}/order-management-feature-overview/order-management-feature-overview.html) + [Inventory Management](/docs/pbc/all/warehouse-management-system/{{page.version}}/base-shop/inventory-management-feature-overview.html) feature into a Spryker project.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-inventory-management-feature.md:4]:
```suggestion
This document describes how to integrate the [Order Management](/docs/scos/user/features/{{page.version}}/order-management-feature-overview/order-management-feature-overview.html) + [Inventory Management](/docs/pbc/all/warehouse-management-system/{{page.version}}/base-shop/inventory-management-feature-overview.html) features into a Spryker project.
```

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 2]
```diff
@@ -979,7 +979,40 @@ Make sure that during checkout on the Shipment step, you can only see shipment m
 
 {% endinfo_block %}
 
-6. To enable the Backend API, register these plugins:
+6. Configure shipment type validation plugin:
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:982]:
```suggestion
6. Configure the shipment type validation plugin:
```

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 3]
```diff
@@ -1076,7 +1109,32 @@ Ensure that the following modules have been installed:
...
+
+2. Import data:
+
+```bash
+console data:import glossary
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that in the database the configured data has been added to the `spy_glossary` table.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:1133]:
```suggestion
Make sure that the configured data has been added to the `spy_glossary` table in the database.
```

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+
+
+This document describes how to integrate the [Order Management](/docs/scos/user/features/{{page.version}}/order-management-feature-overview/order-management-feature-overview.html) + [Shipment](/docs/pbc/all/carrier-management/{{page.version}}/base-shop/shipment-feature-overview.html) feature into a Spryker project.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:3]:
```suggestion
This document describes how to integrate the [Order Management](/docs/scos/user/features/{{page.version}}/order-management-feature-overview/order-management-feature-overview.html) + [Shipment](/docs/pbc/all/carrier-management/{{page.version}}/base-shop/shipment-feature-overview.html) features into a Spryker project.
```

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+Make sure that the following module has been installed:
+
+| MODULE             | EXPECTED DIRECTORY                 |
+|--------------------|------------------------------------|
+| SalesShipmentType  | vendor/spryker/sales-shipment-type |
+
+{% endinfo_block %}
+
+### 2) Set up database schema and transfer objects
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:46]:
```suggestion
### 2) Set up the database schema and transfer objects
```

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+| TRANSFER          | TYPE  | EVENT   | PATH                                                    |
+|-------------------|-------|---------|---------------------------------------------------------|
+| SalesShipmentType | class | created | src/Generated/Shared/Transfer/SalesShipmentTypeTransfer |
+
+{% endinfo_block %}
+
+### 3) Set up behavior
+
+Enable the following plugins.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:74]:
```suggestion
Enable the following plugins:
```

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        return [
+            new ShipmentTypeOrderItemsPostSavePlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that when you place an order selected shipment type is persisted to `spy_sales_shipment_type` and `spy_sales_shipment.fk_sales_shipment_type` is updated.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:107]:
```suggestion
Make sure that when you place an order, the selected shipment type is persisted to `spy_sales_shipment_type` and `spy_sales_shipment.fk_sales_shipment_type` is updated.
```

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+| NAME             | VERSION          | INTEGRATION GUIDE                                                                                                                                                                   |
+|------------------|------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Order Management | {{page.version}} | [Order Management feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/order-management-feature-integration.html)                                        |
+| Shipment         | {{page.version}} | [Shipment feature integration](/docs/pbc/all/carrier-management/{{page.version}}/unified-commerce/enhanced-click-and-collect/install-and-upgrade/install-the-shipment-feature.html) |
+
+
+## 1) Install the required modules using Composer
+
+```bash
+composer require spryker/sales-shipment-type:"^0.1.0" --update-with-dependencies
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:33]: It's a part of shipment feature IG.

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 9]
```diff
@@ -1076,7 +1109,32 @@ Ensure that the following modules have been installed:
...
+### 2) Add translations
+
+Add translations as follows:
+
+1.  Append glossary according to your configuration:
+
+**src/data/import/glossary.csv**
+
+```csv
+shipment_type.checkout.validation.error,Selected delivery type "%name%" is not available,en_US
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:1121]:
```suggestion
shipment_type_cart.checkout.validation.error,Selected delivery type "%name%" is not available,en_US
```
Move it to `shipment + cart` feature IG.

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 10]
```diff
@@ -979,7 +979,40 @@ Make sure that during checkout on the Shipment step, you can only see shipment m
 
 {% endinfo_block %}
 
-6. To enable the Backend API, register these plugins:
+6. Configure the shipment type validation plugin:
+
+| PLUGIN                                 | SPECIFICATION                                                                                               | PREREQUISITES | NAMESPACE                                                                                         |
+|----------------------------------------|-------------------------------------------------------------------------------------------------------------|---------------|---------------------------------------------------------------------------------------------------|
+| ShipmentTypeCheckoutPreConditionPlugin | Validates if selected shipment type have relation to selected shipment method, current store and is active. |               | Spryker\Zed\ShipmentTypeCart\Communication\Plugin\Checkout\ShipmentTypeCheckoutPreConditionPlugin |
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:986]: Move it to `shipment + cart` feature IG.

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+class SalesDependencyProvider extends SprykerSalesDependencyProvider
+{
+    /**
+     * @return list<\Spryker\Zed\SalesExtension\Dependency\Plugin\OrderItemsPostSavePluginInterface>
+     */
+    protected function getOrderItemsPostSavePlugins(): array
+    {
+        return [
+            new ShipmentTypeOrderItemsPostSavePlugin(),
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:99]: ShipmentTypeCheckoutDoSaveOrderPlugin

File: _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md (hunk 8 above)
Comment (AsonUnique) [reply _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:33]: We discussed with @abitskil that we will create a separate IG for this. 
I don't mind moving it, but we need to discuss it first

Comment (abitskil) [reply _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md:33]: Let's add this to the shipment IG.

Comment (AsonUnique) [reply _includes/pbc/all/install-features/202400.0/install-the-order-management-shipment-feature.md]: Moved to main IG

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 12]
```diff
@@ -22,9 +22,10 @@ Follow the steps below to install the Shipment feature core.
...
-|--------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------|
-| Spryker Core | {{page.version}} | [Spryker Core feature integration](/docs/pbc/all/miscellaneous/{{page.version}}/install-and-upgrade/install-features/install-the-spryker-core-feature.html) |  |
+| NAME             | VERSION          | INTEGRATION GUIDE                                                                                                                                           |
+|------------------|------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Spryker Core     | {{page.version}} | [Spryker Core feature integration](/docs/pbc/all/miscellaneous/{{page.version}}/install-and-upgrade/install-features/install-the-spryker-core-feature.html) |
+| Order Management | {{page.version}} | [Order Management feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/order-management-feature-integration.html)                |
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:28]: For shipment it's not a mandatory feature, we just use the sales module as a dependency but we don't need to integrate it fully.

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md [hunk 13]
```diff
@@ -979,7 +983,47 @@ Make sure that during checkout on the Shipment step, you can only see shipment m
 
 {% endinfo_block %}
 
-6. To enable the Backend API, register these plugins:
+6. Configure shipment type order saver plugins:
```
Comment (dmiseev) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-feature.md:986]: `Shipment + Order management feature` IG

**Review by dmiseev — APPROVED** [review]

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+
+To start feature integration, overview, and install the necessary features:
+
+| NAME         | VERSION          |
+|--------------|------------------|
+| Shipment     | {{page.version}} |
+| Cart         | {{page.version}} |
+| Prices       | {{page.version}} |
+
+### 1) Install the required modules using Composer
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md:15]:
```suggestion
### Install the required modules using Composer
```

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+Make sure that the following modules have been installed:
+
+| MODULE                | EXPECTED DIRECTORY                     |
+|-----------------------|----------------------------------------|
+| ShipmentCartConnector | vendor/spryker/shipment-cart-connector |
+| ShipmentTypeCart      | vendor/spryker/shipment-type-cart      |
+
+{% endinfo_block %}
+
+### 2) Set up transfer objects
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md:33]:
```suggestion
### Set up transfer objects
```

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+| ShipmentTypeCollection    | class | Created | src/Generated/Shared/Transfer/ShipmentTypeCollectionTransfer    |
+| ShipmentTypeConditions    | class | Created | src/Generated/Shared/Transfer/ShipmentTypeConditionsTransfer    |
+| ShipmentTypeCriteria      | class | Created | src/Generated/Shared/Transfer/ShipmentTypeCriteriaTransfer      |
+| CheckoutResponse          | class | Created | src/Generated/Shared/Transfer/CheckoutResponseTransfer          |
+| CheckoutError             | class | Created | src/Generated/Shared/Transfer/CheckoutErrorTransfer             |
+| Store                     | class | Created | src/Generated/Shared/Transfer/StoreTransfer                     |
+
+{% endinfo_block %}
+
+### 3) Set up behavior
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md:71]:
```suggestion
### Set up behavior
```

File: _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md [hunk 17]
```diff
@@ -0,0 +1,193 @@
...
+### Prerequisites
+
+To start feature integration, integrate the required features:
+
+| NAME     | VERSION          | 
+|----------|------------------|
+| Shipment | {{page.version}} |
+| Cart     | {{page.version}} |
+
+### 2) Add translations
```
Comment (AlexSlawinski) [inline _includes/pbc/all/install-features/202400.0/install-the-shipment-cart-feature.md:170]:
```suggestion
### Add translations
```

**Review by AlexSlawinski — APPROVED** [review]

**Pull request merged**